lib/pomelo_ex/general/users/create_user.ex

```diff
@@ -1,7 +1,39 @@
 defmodule PomeloEx.General.Users.CreateUser do
   @moduledoc false
+  use TypedEctoSchema
 
-  def execute(payload) do
-    raise "Not implemented #{__MODULE__} payload:" <> inspect(payload)
+  alias PomeloEx.General.Authorization
+  alias PomeloEx.Types.General.Users.CreateUserType
+
+  @idempotency_key_length 14
+
+  def execute(%CreateUserType{} = payload) do
+    http_client = Application.get_env(:pomelo_ex, :http_adapter)
+    url = Application.get_env(:pomelo_ex, :url)
+    headers = build_headers()
+    body = Jason.encode!(payload)
+
+    http_client.post("#{url}/users/v1/", body, headers)
+  end
+
+  defp build_headers do
+    {:ok, %HTTPoison.Response{status_code: 200, body: body}} = Authorization.request_token()
```

Comment:
I would need to pass this token as a parameter to the execute function. The idea behind this token is for the client to reuse it until it expires; however, in this case, a new token is being generated with each request.